cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/service/http/HttpHandlerGeneratorTest.java

```diff
@@ -727,5 +728,15 @@ public void deleteSecureData(String namespace, String name) throws Exception {
     public SecureStoreData getSecureData(String namespace, String name) throws Exception {
       return null;
     }
+
+    @Override
+    public void execute(TxRunnable runnable) throws TransactionFailureException {
+      // nop-op
```

Comment:
This cannot be no-op. Need to call `runnable.run` with a no-op `DatasetContext`. 

Also, it may be good to add more tests in this test class for different combination of transaction policy (handler, content producer/consumer), as this class is intended to cover the correctness of the bytecode generation and this test class should runs much faster than test framework test. Tests in `TestFrameworkTest` is more for integrated type of tests (e.g. dataset context are wired up correctly).